Source/Core/DolphinQt2/QtUtils/Bind.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <functional>
+#include <type_traits>
+
+#include <QAbstractSlider>
+#include <QCheckBox>
+#include <QLabel>
+#include <QSpinBox>
+
+template <typename T>
+struct MemberFunctionArgType;
+
+template <typename R, typename C, typename A>
+struct MemberFunctionArgType<R (C::*)(A)>
+{
+  using ArgType = A;
+};
+
+template <typename R, typename C, typename A>
+struct MemberFunctionArgType<R (C::*const)(A)>
+{
+  using ArgType = A;
+};
+
+template <typename ControlClass, typename EnableIf = void>
+struct BindableControlTrait;
+
+template <>
+struct BindableControlTrait<QCheckBox>
+{
+  static constexpr auto SetFunc = &QCheckBox::setChecked;
+  static constexpr auto SignalFunc = &QCheckBox::toggled;
+};
+
+template <>
+struct BindableControlTrait<QComboBox>
+{
+  static constexpr auto SetFunc = &QComboBox::setCurrentIndex;  // TODO: handle out-of-range values
+  static constexpr auto SignalFunc = static_cast<void (QComboBox::*)(int)>(&QComboBox::activated);
+};
+
+void ButtonGroupSetChecked(QButtonGroup* button_group, int id)
```

Comment:
This should likely be inlined or have the definition moved to a cpp file. The include for QButtonGroup is also missing.